### ISSUE TITLE
Changing Allowed Area Freezes Pawns

### ISSUE THREAD

**Issue by AlphaAtlas** [rheirman/GiddyUpRideAndRoll#16]
Changing the allowed area of both a mountable animal and a colonist freezes the colonist, and spams the error log. 

Steps to reproduce: 

- Quickstart a new game, tame a mountable animal close to (in mountable range of) a colonist.

- Expand "Area 1" to an area far away from the colonists and ride-able animal.

- Pause the game, set the allowed area for both the colonist and the animal to "Area 1." The error log immediately pops up when paused, and only stops when the colonist is drafted or the animal moves out of mounting range. 

For real world context, I always run into this issue when raids arrive, as I change the allowed area to inside the base while colonists are doing work with their animals in the field.

Here's a sample log. I went ahead and enabled job logging for one of the colonists: 

<details>
  <summary>Click to Expand Error Log</summary>
  
  ```
Log uploaded on Wednesday, September 04, 2019, 11:16:37 PM
Loaded mods:
Core: (no assemblies)
HugsLib[ov:6.1.3]: 0Harmony(1.2.0.1), HugsLib(av:1.0.0.0,fv:6.1.3)
Giddy-up! Core[ov:1.1.6]: GiddyUpCore(0.0.0.0)
Giddy-up! Battle Mounts[ov:1.0.2]: Battlemounts(0.0.0.0), DependencyChecker(1.0.0.0)
Giddy-up! Caravan[ov:1.1.0]: DependencyChecker(1.0.0.0), GiddyUpCaravan(0.0.0.0)
Giddy-up! Ride and Roll[ov:1.2.0]: DependencyChecker(1.0.0.0), GiddyUpRideAndRoll(0.0.0.0)

...

[HugsLib] Quickstarter generating map with scenario: Crashlanded

Unloading 3 Unused Serialized files (Serialized files now loaded: 0)

Unloading 29 unused Assets to reduce memory usage. Loaded Objects now: 15826.
Total: 316.636261 ms (FindLiveObjects: 2.219066 ms CreateObjectMapping: 2.452266 ms MarkObjects: 311.701416 ms  DeleteObjects: 0.254711 ms)

Unable to get latest version information from rimworldgame.com. (Couldn't resolve host name)

Initializing new game with mods Core, 818773962, 1216999901, 1217001091, 1253251478, 1331961995

Cleaned up 0 depricated records from Giddy-up!

Unloading 1 Unused Serialized files (Serialized files now loaded: 0)

Unloading 0 unused Assets to reduce memory usage. Loaded Objects now: 21367.
Total: 482.316315 ms (FindLiveObjects: 4.211778 ms CreateObjectMapping: 3.372844 ms MarkObjects: 474.541504 ms  DeleteObjects: 0.183822 ms)

Odette started 10 jobs in one tick. newJob=Mount (Job_186) A=Thing_Muffalo48272 jobGiver=Verse.AI.ThinkNode_QueuedJob jobList=(Mount (Job_176) A=Thing_Muffalo48272) (Mount (Job_177) A=Thing_Muffalo48272) (Mount (Job_178) A=Thing_Muffalo48272) (Mount (Job_179) A=Thing_Muffalo48272) (Mount (Job_180) A=Thing_Muffalo48272) (Mount (Job_181) A=Thing_Muffalo48272) (Mount (Job_182) A=Thing_Muffalo48272) (Mount (Job_183) A=Thing_Muffalo48272) (Mount (Job_184) A=Thing_Muffalo48272) (Mount (Job_185) A=Thing_Muffalo48272) (Mount (Job_186) A=Thing_Muffalo48272)  lastJobGiver=Verse.AI.ThinkNode_QueuedJob

Jimbo started 10 jobs in one tick. newJob=Mount (Job_200) A=Thing_Muffalo48272 jobGiver=Verse.AI.ThinkNode_QueuedJob jobList=(Mount (Job_190) A=Thing_Muffalo48272) (Mount (Job_191) A=Thing_Muffalo48272) (Mount (Job_192) A=Thing_Muffalo48272) (Mount (Job_193) A=Thing_Muffalo48272) (Mount (Job_194) A=Thing_Muffalo48272) (Mount (Job_195) A=Thing_Muffalo48272) (Mount (Job_196) A=Thing_Muffalo48272) (Mount (Job_197) A=Thing_Muffalo48272) (Mount (Job_198) A=Thing_Muffalo48272) (Mount (Job_199) A=Thing_Muffalo48272) (Mount (Job_200) A=Thing_Muffalo48272)  lastJobGiver=Verse.AI.ThinkNode_QueuedJob

Julie started 10 jobs in one tick. newJob=Mount (Job_214) A=Thing_Muffalo48272 jobGiver=Verse.AI.ThinkNode_QueuedJob jobList=(Mount (Job_204) A=Thing_Muffalo48272) (Mount (Job_205) A=Thing_Muffalo48272) (Mount (Job_206) A=Thing_Muffalo48272) (Mount (Job_207) A=Thing_Muffalo48272) (Mount (Job_208) A=Thing_Muffalo48272) (Mount (Job_209) A=Thing_Muffalo48272) (Mount (Job_210) A=Thing_Muffalo48272) (Mount (Job_211) A=Thing_Muffalo48272) (Mount (Job_212) A=Thing_Muffalo48272) (Mount (Job_213) A=Thing_Muffalo48272) (Mount (Job_214) A=Thing_Muffalo48272)  lastJobGiver=Verse.AI.ThinkNode_QueuedJob

424 Julie: Job expire

424 Julie: EndCurrentJob Wait (Job_215) condition=Succeeded curToil=0

424 Julie: CleanupCurrentJob Wait condition Succeeded

424 Julie: StartJob [Wait_MaintainPosture (Job_318)] lastJobEndCondition=None, jobGiver=, cancelBusyStances=False

Odette started 10 jobs in one tick. newJob=Mount (Job_329) A=Thing_Muffalo48272 jobGiver=Verse.AI.ThinkNode_QueuedJob jobList=(Mount (Job_319) A=Thing_Muffalo48272) (Mount (Job_320) A=Thing_Muffalo48272) (Mount (Job_321) A=Thing_Muffalo48272) (Mount (Job_322) A=Thing_Muffalo48272) (Mount (Job_323) A=Thing_Muffalo48272) (Mount (Job_324) A=Thing_Muffalo48272) (Mount (Job_325) A=Thing_Muffalo48272) (Mount (Job_326) A=Thing_Muffalo48272) (Mount (Job_327) A=Thing_Muffalo48272) (Mount (Job_328) A=Thing_Muffalo48272) (Mount (Job_329) A=Thing_Muffalo48272)  lastJobGiver=Verse.AI.ThinkNode_QueuedJob

Jimbo started 10 jobs in one tick. newJob=Mount (Job_341) A=Thing_Muffalo48272 jobGiver=Verse.AI.ThinkNode_QueuedJob jobList=(Mount (Job_331) A=Thing_Muffalo48272) (Mount (Job_332) A=Thing_Muffalo48272) (Mount (Job_333) A=Thing_Muffalo48272) (Mount (Job_334) A=Thing_Muffalo48272) (Mount (Job_335) A=Thing_Muffalo48272) (Mount (Job_336) A=Thing_Muffalo48272) (Mount (Job_337) A=Thing_Muffalo48272) (Mount (Job_338) A=Thing_Muffalo48272) (Mount (Job_339) A=Thing_Muffalo48272) (Mount (Job_340) A=Thing_Muffalo48272) (Mount (Job_341) A=Thing_Muffalo48272)  lastJobGiver=Verse.AI.ThinkNode_QueuedJob

425 Julie: Job expire

425 Julie: EndCurrentJob Wait_MaintainPosture (Job_318) condition=Succeeded curToil=0

425 Julie: CleanupCurrentJob Wait_MaintainPosture condition Succeeded

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_343) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_343) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_343) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_344) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_344) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_344) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_345) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_345) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_345) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_346) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_346) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_346) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_347) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_347) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_347) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_348) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_348) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_348) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_349) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_349) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_349) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_350) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_350) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_350) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_351) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_351) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_351) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

425 Julie: StartJob [Mount (Job_352) A=Thing_Muffalo48272] lastJobEndCondition=None, jobGiver=Verse.AI.ThinkNode_QueuedJob, cancelBusyStances=False

425 Julie: JobDriver_Mount ends current job Mount (Job_352) A=Thing_Muffalo48272 because of globalFailConditions[1]

425 Julie: EndCurrentJob Mount (Job_352) A=Thing_Muffalo48272 condition=Incompletable curToil=0

425 Julie: CleanupCurrentJob Mount condition Incompletable

425 Julie: TryFindAndStartJob

425 Julie:    Returning queued job: Goto (Job_202) A=(154, 0, 182)

Julie started 10 jobs in one tick. newJob=Mount (Job_353) A=Thing_Muffalo48272 jobGiver=Verse.AI.ThinkNode_QueuedJob jobList=(Mount (Job_343) A=Thing_Muffalo48272) (Mount (Job_344) A=Thing_Muffalo48272) (Mount (Job_345) A=Thing_Muffalo48272) (Mount (Job_346) A=Thing_Muffalo48272) (Mount (Job_347) A=Thing_Muffalo48272) (Mount (Job_348) A=Thing_Muffalo48272) (Mount (Job_349) A=Thing_Muffalo48272) (Mount (Job_350) A=Thing_Muffalo48272) (Mount (Job_351) A=Thing_Muffalo48272) (Mount (Job_352) A=Thing_Muffalo48272) (Mount (Job_353) A=Thing_Muffalo48272)  lastJobGiver=Verse.AI.ThinkNode_QueuedJob

425 Julie: StartJob [Wait (Job_354)] lastJobEndCondition=None, jobGiver=, cancelBusyStances=True
  ```
</details>

~~Now I don't know much about C# RimWorld modding, so I'm in _way_ over my head here, but it looks like this line in JobDriver_Mounted.cs is repeatedly canceling the mounting job:~~

~~else if (!__instance.Rider.pather.Moving)~~

~~And the colonist tries to pick up another mounting job each time its canceled.~~

nvm, just saw the not

**Comment by jinlan** [comment]
Same.

Reproduce:
1 Restrict your pawn to a zone.
2 Move that pawn out of the zone.
3 If there is a mountable animal near by, then he will try to mount it and ride back to the zone. 
3? Maybe animal should out of its restricted zone too, I'm not sure.
4 10-job-one-tick error comes up.

```
Wasp started 10 jobs in one tick. newJob=Mount (Job_771572) A=Thing_Bear_Grizzly67245 jobGiver=Verse.AI.ThinkNode_QueuedJob jobList=(Wait (Job_771559)) (Mount (Job_771563) A=Thing_Bear_Grizzly67245) (Mount (Job_771564) A=Thing_Bear_Grizzly67245) (Mount (Job_771565) A=Thing_Bear_Grizzly67245) (Mount (Job_771566) A=Thing_Bear_Grizzly67245) (Mount (Job_771567) A=Thing_Bear_Grizzly67245) (Mount (Job_771568) A=Thing_Bear_Grizzly67245) (Mount (Job_771569) A=Thing_Bear_Grizzly67245) (Mount (Job_771570) A=Thing_Bear_Grizzly67245) (Mount (Job_771571) A=Thing_Bear_Grizzly67245) (Mount (Job_771572) A=Thing_Bear_Grizzly67245)  lastJobGiver=Verse.AI.ThinkNode_QueuedJob
Verse.Log:Error(String, Boolean)
Verse.AI.JobUtility:TryStartErrorRecoverJob_Patch1(Pawn, String, Exception, JobDriver)
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.JobDriver:EndJobWith(JobCondition)
Verse.AI.JobDriver:CheckCurrentToilEndOrFail()
Verse.AI.JobDriver:TryActuallyStartNextToil()
Verse.AI.JobDriver:ReadyForNextToil()
Verse.AI.Pawn_JobTracker:StartJob_Patch2(Object, Job, JobCondition, ThinkNode, Boolean, Boolean, ThinkTreeDef, Nullable`1, Boolean)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
RimWorld.Pawn_DraftController:set_Drafted_Patch4(Object, Boolean)
RimWorld.<GetGizmos>c__Iterator0:<>m__0()
Verse.Command_Toggle:ProcessInput(Event)
Verse.GizmoGridDrawer:DrawGizmoGrid(IEnumerable`1, Single, Gizmo&)
RimWorld.InspectGizmoGrid:DrawInspectGizmoGridFor(IEnumerable`1, Gizmo&)
RimWorld.MainTabWindow_Inspect:DrawInspectGizmos()
RimWorld.InspectPaneUtility:ExtraOnGUI(IInspectPane)
RimWorld.MainTabWindow_Inspect:ExtraOnGUI()
Verse.WindowStack:WindowStackOnGUI()
RimWorld.UIRoot_Play:UIRootOnGUI()
Verse.Root:OnGUI()
```

**Comment by 20k** [comment]
I think a similar issue to this lead to crashes in the multiplayer mod, we pinned it down to area changes with giddy up ride and roll

**Comment by Gusgus01** [comment]
This fixes the issue, but in a way that I am not necessarily happy with. It just short circuits the decision to mount when the circumstances that causes the issue comes up. It appears that there is a nasty loop when you restrict the area and the colonists decide to use the mounts to get to the area.

Stepping through the code using dnSpy, I see something calls EndCurrentJob(), possibly Pawn_PlayerSettings.set_AreaRestriction(), then it decides to assign the Goto job, then this mod changes that decision to Mount. And then all that repeats over and over.

**Comment by Gusgus01** [comment]
A better solution might be to patch the calling code ie set_AreaRestriction() or patch the decision in EndCurrentJob(), but those would expand the mod's functionality and would be harder to get correct.

**Comment by rheirman** [comment]
I'll look into it now. Thanks for all the info. I wouldn't want a solution that restricts pawn behaviour in an unpredictable way. The PR is indeed too restrictive. I'll see if I can come up with something.

**Comment by rheirman** [comment]
After fiddling with this issue for quite a while, I found the source of the problem. 

Apparently in Jobdriver_Mount, the following line is present in MakeNewToils:
this.FailOnDespawnedNullOrForbidden(TargetIndex.A);

The code that replaces the GoTo job with the Mount job does however not take into account restricted areas, resulting in the pawn endlessly trying to attempt to mount the animal and failing because the animal is outside the restricted area of the pawn.   

So the mount job fails when the target which should be mounted is outside the allowed area of the pawn. I simply added the following code to the method named Pawn_Jobtracker_DetermineNextJob.AnimalNotAvailable, and passed the rider as an argument in that function:

```
            if (animal.IsForbidden(rider))
            {
                return true;
            }
```

This solves it, and also prevents pawns from leaving their allowed area to mount an animal, which makes a lot of sense.
